pkg/controllers/awsloadbalancercontroller/rbac_clusterrole.go

```diff
@@ -0,0 +1,131 @@
+package awsloadbalancercontroller
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	albo "github.com/openshift/aws-load-balancer-operator/api/v1alpha1"
+)
+
+// ensureClusterRoles creates and maintains clusterroles required by the aws-load-balancer controller.
+func (r *AWSLoadBalancerControllerReconciler) ensureClusterRoles(ctx context.Context, controller *albo.AWSLoadBalancerController) error {
+	reqLogger := log.FromContext(ctx)
+
+	desired := desiredClusterRoles(ctx)
+
+	for _, obj := range desired {
+		if err := controllerutil.SetControllerReference(controller, obj, r.Scheme); err != nil {
+			return fmt.Errorf("failed to set the controller reference for clusterroles %s : %w", obj.Name, err)
+		}
+	}
+
+	exist, current, err := r.currentClusterRoles(ctx, desired)
+	if err != nil {
+		return fmt.Errorf("failed to fetch current clusterroles: %w", err)
+	}
+
+	if !exist {
+		if err := r.createClusterRoles(ctx, desired); err != nil {
+			return err
+		}
+		_, current, err = r.currentClusterRoles(ctx, desired)
+		if err != nil {
+			return fmt.Errorf("failed to fetch created clusterroles: %w", err)
+		}
+		reqLogger.Info("created clusterroles", "count", len(current))
+	}
+
+	if updated, err := r.updateClusterRoles(ctx, current, desired); err != nil {
+		return fmt.Errorf("failed to update clusterroles: %w", err)
+	} else if updated {
+		_, _, err = r.currentClusterRoles(ctx, desired)
+		return err
+	}
+
+	return nil
+}
+
+// updateClusterRoles updates the current clusterroles and returns a flag to denote if the update was done.
+func (r *AWSLoadBalancerControllerReconciler) updateClusterRoles(ctx context.Context, current map[string]*rbacv1.ClusterRole, desired []*rbacv1.ClusterRole) (bool, error) {
```

Comment:
The _updated_ flag is not required because the re-read in the calling function is really not required.